### ISSUE TITLE
spatial stage test problem

### ISSUE THREAD

**Issue by LeRoii** [mwaseema/Drone-Detection#5]
hello, I followed the instructions in README.md and finished training the spatial model, I took 10-20.mov of NPS dataset as test set and others as train set.
then I ran test_and_score.py but the result seemed incorrect, in the output detection folder, all the images are black and there is no mask. Is there something I did wrong?
And what is the relation between spatial stage and temporal stage, can I just run one of them independently?
 Thanks in advance.

**Comment by mwaseema** [comment]
Hi LeRoii,
There must've been some problem which caused this. Can you please elaborate on how you did what? That'll help me understand the problem and guide you.

You must've used `tools/video_to_frames_and_masks.py` script for extracting frames from videos data.
Please make sure you've set `foreground_in_mask` to `1` for training data and to `255` for testing data.

Thanks to you. I've also updated this in the readme.

The temporal stage uses the output of the spatial stage as a candidate region for the availability of the object. You'll need output from stage one to actually benefit from temporal properties.

Regards,

**Comment by LeRoii** [comment]
> 
> You must've used `tools/video_to_frames_and_masks.py` script for extracting frames from videos data. Please make sure you've set `foreground_in_mask` to `1` for training data and to `255` for testing data.
> 
hi mwaseema, thanks for your reply, it is `tools/video_to_frames_and_masks.py` script that caused the problem, I set `foreground_in_mask` to 1 and regenerated training set, then I trained for 80+ epoch by now and ran `test_and_score.py` for one image, the result seemed good.
but the training loss is relative high, it's around 1000 for 80+ epoch, and it seems that the loss would not go down any further, 
is this normal?

![Screenshot from 2021-12-16 09-44-13](https://user-images.githubusercontent.com/23433742/146292682-c217af80-8fad-48d3-b8a8-3eb45734404d.png)

**Comment by mwaseema** [comment]
Thanks, LeRoii for reporting back. Happy to learn the problem was fixed. And the loss part is normal.
